### ISSUE TITLE
About the weight selection of Route Loss and the calculation of M

### ISSUE THREAD

**Issue by lyb18758** [bytedance/DreamO#90]
Hello author, thanks for such a great job of open source! We want to train a new model based on your work. Now there are two more questions I would like you to answer.
1. When calculating the M of the condition, do you calculate and output all the 19-layer DoubleStream and 38-layer SingleStream output of FLUX for calculating the L_route?
2. How to set the weight of the L_route?
Looking forward to hearing from you。

**Comment by MC-E** [comment]
Yes, we calculate M in all layers. The value of L_route is comparable to that of L_diff, and we maintain a 1:1 weighting.

**Comment by lyb18758** [comment]
> Yes, we calculate M in all layers. The value of L_route is comparable to that of L_diff, and we maintain a 1:1 weighting.

Thanks for your answer. I still have a question, at what stage of training did L_route add it? Is it training from the very beginning or on top of an existing checkpoint?

**Comment by MC-E** [comment]
> > Yes, we calculate M in all layers. The value of L_route is comparable to that of L_diff, and we maintain a 1:1 weighting.
> 
> Thanks for your answer. I still have a question, at what stage of training did L_route add it? Is it training from the very beginning or on top of an existing checkpoint?

Except for the final quality fine-tuning stage, all other stages should use:)

**Comment by lyb18758** [comment]
> > > Yes, we calculate M in all layers. The value of L_route is comparable to that of L_diff, and we maintain a 1:1 weighting.
> > 
> > 
> > Thanks for your answer. I still have a question, at what stage of training did L_route add it? Is it training from the very beginning or on top of an existing checkpoint?
> 
> Except for the final quality fine-tuning stage, all other stages should use:)

Thank you for your response. We noticed that the paper used subject200k to stitch images into two columns for training. How exactly is it trained? Be specific：
1.How are the target image and reference image set up? 
2.We speculate that the stitched image is used as the ground truth (GT), with the GT masks extracted separately. The reference image consists of half of the GT with the other half filled in white, ensuring that the image size matches the GT to avoid errors during attention loss. Is this how the training is done?
3.We noticed that the original subject200k dataset is a two-column image stitching, should we directly use the original data or re-split and re-construct to get the new dataset?
Hope for your answer！
